pkg/actions/lua/lakefs/catalogexport/delta_exporter.lua

```diff
@@ -0,0 +1,134 @@
+local lakefs = require("lakefs")
+local formats = require("formats")
+local pathlib = require("path")
+local json = require("encoding/json")
+local utils = require("lakefs/catalogexport/internal")
+
+local function delta_log_entry_key_generator()
+    local current = 0
+    return function()
+        local delta_log_entry_length = 20
+        local key = tostring(current)
+        local padding_length = delta_log_entry_length - key:len()
+        local padded_key = ""
+        for _ = 1, padding_length do
+            padded_key = padded_key .. "0"
+        end
+        padded_key = padded_key .. key
+        current = current + 1
+        return padded_key
+    end
+end
+
+local function get_delta_client(key, secret, region)
+    return formats.delta_client("s3", key, secret, region)
+end
```

Comment:
Using this function instead of directly calling `formats.delta_client("s3", key, secret, region)` is done because the delta client implementation uses the S3 gateway to communicate with lakeFS.
Once it is possible to use the lakeFS API directly there will be a meaning to the actual storage type.